src/sqlfluff/rules/structure/ST10.py

```diff
@@ -0,0 +1,235 @@
+"""Implementation of Rule ST10."""
+
+from typing import Iterator, List, Tuple, cast
+
+from sqlfluff.core.parser.segments import BaseSegment
+from sqlfluff.core.rules import BaseRule, LintResult, RuleContext
+from sqlfluff.core.rules.crawlers import SegmentSeekerCrawler
+from sqlfluff.dialects.dialect_ansi import ObjectReferenceSegment
+
+
+class UnqualifiedReferenceError(ValueError):
+    """Custom exception for signalling when a reference is unqualified."""
+
+
+class Rule_ST10(BaseRule):
+    """Joined table not referenced in query.
+
+    This rule will check if there are any tables that are referenced in the
+    ``FROM`` or ``JOIN`` clause of a ``SELECT`` statement, but where no
+    columns from that table are referenced in the any of the other clauses.
+
+    This rule relies on all of the column references in the ``SELECT``
+    statement being qualified with at least the table name, and so is
+    designed to work alongside :sqlfluff:ref:`references.qualification`
+    (:sqlfluff:ref:`RF02`).
+
+    This rule does not propose a fix, because it assumes that it an unused
```

Comment:
Makes sense. I could imagine unusual cases where someone joins with a table simply for filtering purposes, and there's nothing "wrong" with that. (Although one could argue in that case `WHERE EXISTS` with a sub-select would be clearer about the _intent_.

Reply:
To address this issue, I've added an `allowlist` of specific keywords, which includes `INNER` to prevent this situation. By making it configurable, people can make it more or less aggressive as they prefer.